experimental-examples/kitchen-sink/tina/custom-component-reference-select/reference-field-schema.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import AuthorCollectionCustomReference from '../../component/custom-reference-select-author'
+import PostCollectionCustomReference from '../../component/custom-reference-select-post'
+import { CollectionProps, COLLECTIONS } from './model'
+
+const referenceField = {
+  label: 'Author',
+  name: 'author',
+  type: 'reference',
+  ui: {
+    selectComponents: (props: CollectionProps, filepath: string) => {
```

Comment:
This should `selectComponent`. If you fix that you'll see the types from [here](https://github.com/tinacms/tinacms/pull/4732/files#diff-88b24591b0c36f508270e9ae261e94ea2770198c387d66a95b2fe1620810bde3R220). They won't know the shape of the `values`, but you could cast them here. So if you want type safety on these I think it'd make sense to do (assuming the signature I mentioned below is used):

```ts
    selectComponent: (args) => {
      const props = args.values as CollectionProps
      const filename = args._sys.filepath
    
```

Reply:
Done - the current example here do the type safety by defined the interface for the param. They both works the same i guess